src/shared/gpt.c

```diff
@@ -209,11 +209,34 @@ const char *gpt_partition_type_uuid_to_string_harder(
 }
 
 int gpt_partition_type_from_string(const char *s, GptPartitionType *ret) {
+        _cleanup_free_ char *alias = NULL;
+        const char *p, *subst = NULL;
         sd_id128_t id;
         int r;
 
         assert(s);
 
+        /* Two special cases: alias aarch64 to arm64, and amd64 to x86-64. The DSP mixes debianisms and
+         * CPUisms: for x86, it uses x86 and x86_64, but for aarch64 it uses arm64. This is confusing, and
+         * leads to issues for callers that have to know which -ism to use for which architecture. But we
+         * also don't really want to change the spec and add new partition labels, so add a user-friendly
+         * aliasing here. */
+        if ((p = endswith(s, "aarch64")))
+                subst = "arm64";
+        else if ((p = endswith(s, "amd64")))
+                subst = "x86-64";
```

Comment:
this doesn't work, the string can also show up in the middle of the string (e.g. `root-riscv32-verity-sig`).

i hate this redundancy tbh, noone but debian calls this amd64...

but if you really want this, just add this to gpt_partition_type_table[] table, it can handle non-unique partition type translations anyway. (because on my machine `root` and `root-x86-64` actually map to the same entry)

Reply:
> As the comment mentions, I wanted to ensure that whichever alias is used, the disk is the same. Adding a new entry to the table means the label on disk will be different.

That's not true. Look at it. same uuid can appear multiple times in the table. we *do* *this* *already*.

when i picked the names i picked the names that most sense to me. i really didn't look at debian at all for this.

Reply:
I mean the literal label. If you pass `Type=root-aarch64`, the disk will get a `root-aarch64` label, instead of `root-arm64`.

Reply:
what do you mean by "root-aarch64" label?

there's only one gpt partition type, and we give it the root-arm64 name, but this name doesn't end up on the gpt partition table.

Reply:
it ends in the partition label:

```
$ lsblk -o NAME,TYPE,PARTLABEL
NAME        TYPE PARTLABEL
loop0       loop 
├─loop0p1   part esp
└─loop0p2   part root-arm64
```

What you suggest is the first thing I tried, but then the DDI ends up with a partition label `root-aarch64` which breaks compat with the DPS and any tool that parses the part label